app/components/satis/dropdown/component_controller.js

```diff
@@ -284,7 +284,11 @@ export default class extends ApplicationController {
     option.text = selectedValue
     option.value = selectedValue
     option.setAttribute("selected", true)
-
+    Array.prototype.slice.call(dataDiv.attributes).forEach((attr) => {
```

Comment:
this looks good but fyi the items in hidden select can be added outside the scope of this controller i.e. through populate controller, i would suggest to modify the order_line_controller and do the same there.